server/server.go

```diff
@@ -410,7 +410,7 @@ func (s *server) newErrorEncoderWrapper() kithttp.ErrorEncoder {
 		s.errorEncoder(ctx, responseError, rw)
 
 		// Log the error and its stack. This is really useful for debugging.
-		s.logger.Log("level", "error", "message", "stop endpoint processing due to error", "stack", fmt.Sprintf("%#v", serverError))
+		s.logger.LogCtx(ctx, "level", "error", "message", "stop endpoint processing due to error", "stack", fmt.Sprintf("%#v", microerror.Mask(serverError)))
```

Comment:
`LogCtx` is useful, but additional masking doesn't bring more information. You have that in the `caller`.

The thing I suggested in the PM is getting `err.Error()` output somehow.

Just for reference: 

```
stack=&net.OpError{Op:"write", Net:"tcp", Source:(*net.TCPAddr)(0xc420491a40), Addr:(*net.TCPAddr)(0xc420491a70), Err:(*os.SyscallError)(0xc4201dea40)}
```

I don't see how additional wrapping can give more information here.

One thing that is in my head is we can provide a function like `microerror.PrintStack` which we should use to print errors in logs like `"stack", microerror.PrintStack(err)`. In such a function we could do various things to error printing and fix the issue we have here.

Reply:
Thinking more maybe I'm wrong and masking would print the actual error. But maybe it's good to be ready for such situations in the future.

Reply:
I am not sure, I was under the impression that masking helps with printing the error. Uncertain. 

Otherwise `serverError.Error()` is available if you think that would help.

Reply:
> I am not sure, I was under the impression that masking helps with printing the error. Uncertain.

Maybe it helps. You can test that. What I'd like to have is a general solution to a specific problem.